### DISCUSSION TITLE
Pooling output dimensions confusion

### DISCUSSION THREAD

**Discussion by EngineerKhan** [google/flax#1663] [Q&A] [answered]
I am trying to use flax for max pooling. 

```y = nn.max_pool(x[None,None],window_shape=(2, 2), strides=(2, 2))```

Here `y` is a 6x6 JAX array, but `y` has the shape of (1.0,3,6). 

If I am not wrong, it should have been 3x3 in the last two dimensions. What is wrong there, and how can I fix it?

**Comment by jheek** [comment, marked as answer]
max_pool expects (batch, [spatial dims...], features) so I think what you should do here is replace x[None,None] with x[None, ..., None]

**Reply by EngineerKhan** [reply]
Thanks. It worked!
